test/Microsoft.NET.Build.Tests/GivenThatWeWantToGenerateADepsFileForATool.cs

```diff
@@ -0,0 +1,239 @@
+﻿using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using NuGet.Packaging;
+using NuGet.ProjectModel;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Xml.Linq;
+using Xunit;
+using static Microsoft.NET.TestFramework.Commands.MSBuildTest;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToGenerateADepsFileForATool : SdkTest
+    {
+        [Fact]
+        public void It_creates_a_deps_file_for_the_tool_and_the_tool_runs()
+        {
+            TestProject toolProject = new TestProject()
+            {
+                Name = "TestTool",
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.0",
+                IsExe = true
+            };
+
+            GenerateDepsAndRunTool(toolProject)
+                .Should()
+                .Pass()
+                .And.HaveStdOutContaining("Hello World!");
+        }
+
+        [Fact]
+        public void It_handles_conflicts_when_creating_a_tool_deps_file()
+        {
+            TestProject toolProject = new TestProject()
+            {
+                Name = "DependencyContextTool",
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.0",
+                IsExe = true
+            };
+
+            toolProject.PackageReferences.Add(new TestPackageReference("Microsoft.Extensions.DependencyModel", "2.0.0-preview1-002022", null));
```

Comment:
Since the point is to test conflict resolution, I think we want a fixed version so that we don't roll forward to a version of the library that doesn't need it (because it targets .NET Core / .NET Standard 2.0).